docs/advanced-concepts/composables.md

```diff
@@ -1 +1,33 @@
 # Composables
+
+## useRoute
+
+Returns the current route with built in type guard.
+
+```ts
+import { useRoute } from '@kitbag/router'
+
+const route = useRoute('route.key.here')
+```
+
+the route key argument is what acts as a route guard, providing back the correct types for `params` and `update`. This route key argument is optional on `useRoute`, if omitted will return the same wider type as `Router.route`. The key must match the current route, otherwise router will throw an error.
```

Comment:
Can we expand the type guard part to include how it works if the key is any part of the current route. It can be a parent.